prov/efa/src/rdm/rxr_op_entry.c

```diff
@@ -290,3 +294,632 @@ size_t rxr_tx_entry_num_req(struct rxr_op_entry *tx_entry, int pkt_type)
 	return (total_size - tx_entry->bytes_sent - 1)/tx_entry->max_req_data_size + 1;
 }
 
+/**
+ * @brief handle the situation that an error has happened to an RX (receive) operation
+ *
+ * This function does the following to handle error:
+ *
+ * 1. write an error cq entry to notify application the rx
+ *    operation failed. If write failed, it will write an eq entry.
+ *
+ * 2. increase error counter.
+ *
+ * 3. print warning message about the error with self and peer's
+ *    raw address.
+ *
+ * 4. release resources owned by the RX entry, such as unexpected
+ *    packet entry, because the RX operation is aborted.
+ *
+ * 5. remove the rx_entry from queued rx_entry list for the same reason.
+ *
+ * Note, It will NOT release the rx_entry because it is still possible to receive
+ * packet for this rx_entry.
+ *
+ * @param[in]	rx_entry	rx_entry encoutnered error
+ * @param[in]	err		positive libfabric error code
+ * @param[in]	prov_errno	positive provider specific error code
+ */
+void rxr_rx_entry_handle_error(struct rxr_op_entry *rx_entry, int err, int prov_errno)
```

Comment:
Asking for second opinion about names.

The file name is `rxr_op_entry.c`, but the function names are sometimes `rxr_*x_entry_*`

How about abbreviating `ops_entry` to `ope` for conciseness? Then we can use `rx_ope` and `tx_ope`

This can be done in a separate PR.

Reply:
Yes. There are two types of op_entry: tx_entry and rx_entry. When a function is exclusively applied to a tx_entry or a rx_entry, I used tx/rx in function name to improve readability.

I would be strongly against abbreviating `op_entry` to `ope`. because `ope` is too ambiguous.

Reply:
I think overall `ope` is more beneficial and readable in the long run given that we are already used to `cqe`.

The current naming convention can be improved. There are many different types of 'entries' laying around, e.g. `rxr_pkt_entry`, `fi_cq_err_entry`, `rxr_op_entry`. Especially for `rxr_pkt_entry` vs `rxr_op_entry` - it helps to make them pop out.

Also `xxx_entry` is mouthful to talk about, so I think `ope` would help for verbal communication as well.

Reply:
I think such an abbreviation must be a team decision, and cannot be done in this PR.